docs/development/language-features.md

```diff
@@ -0,0 +1,210 @@
+- `transforms` are the actual deinflection rules
+
+Consider this example:
+
+```js
+// abridged from english-transforms.js
+import { suffixInflection } from "../language-transforms.js";
+
+export const englishTransforms = {
+  language: "en",
+  conditions: {
+    n: {
+      name: "Noun",
+      isDictionaryForm: true,
+      subConditions: ["np", "ns"],
+    },
+    np: {
+      name: "Noun plural",
+      isDictionaryForm: true,
+    },
+    ns: {
+      name: "Noun singular",
+      isDictionaryForm: true,
+    },
+  },
+  transforms: [
+    {
+      name: "plural",
+      description: "Plural form of a noun",
+      rules: [suffixInflection("s", "", ["np"], ["ns"])],
+    },
+  ],
+};
+```
+
+This is a simple example for English, where the only deinflection rule is to remove the "s" from the end of a noun to get the singular form. The `suffixInflection` function is a helper that creates a deinflection rule for a suffix. It takes the suffix to remove, what to replace it with, and the conditions under which the rule can be applied. It is the most common type of deinflection rule across languages.
+
+For the input string "cats", the following strings will be looked up:
+
+- `cats` (no deinflection)
+- `cat` with condition `ns` (deinflected by the `plural` rule)
+
+If the dictionary contains an entry for `cat`, marked as a `n` (noun) it will successfully match the looked up string; since `ns` is a subcondition of `n`, they match.
+
+It is important to note that rules chain together. If the `conditionsOut` of one rule match the `conditionsIn` of another rule, the algorithm will also attempt a lookup with the second rule applied to the output of the first rule. This is indispensable for agglutinative languages, but may cause unexpected behavior. For example, if the `plural` rule in the example were `suffixInflection('s', '', ['n'], ['n'])`. Scanning the word `boss` would also display results for the word `bo` (e.g. the staff) with the `plural` rule applied twice.
+
+A list of the rule chains that were applied is shown in the search results, marked with a 🧩.
+
+The `suffixInflection` is a helper function - you can write more complex rules, using regex and a function for deinflecting. There are examples of this across the language transforms files.
+
+### Text Postprocessors
+
+In special cases, text may need to be modified after deinflection. These work exactly like text preprocessors, but are applied after deinflection. Currently, this is only used for Korean, where the Hangul text is disassembled into jamo during preprocessing, and so must be reassembled after deinflection.
+
+### Text Lookup Worthiness
+
+Some features include checking whether a string is possibly a word in the language. For example, a word written in japanese characters (e.g. "日本語") is not a possible English word, and vice versa for text in the Latin script. To prevent unnecessary lookups, an `isTextLookupWorthy` function can be provided, otherwise all text will be looked up.
```

Comment:
I think I got tripped up being a Japanese learner and assumed the example was for someone learning Japanese so got confused. Suggestion to make it clearer?

```suggestion
Some features include checking whether a string could have a potential entry in a language. For example, trying to look up an English word written with non-Latin characters (e.g. "日本語" in Japanese) will never yield any results. To prevent unnecessary lookups, an `isTextLookupWorthy` function can be provided, otherwise all text will be looked up.
```